src/Microsoft.OData.Core/Json/ODataJsonOutputContext.cs

```diff
@@ -76,8 +76,14 @@ internal sealed class ODataJsonOutputContext : ODataOutputContext
 
                 IJsonWriterFactory jsonWriterFactory = this.Container?.GetService<IJsonWriterFactory>();
 
-#if NETCOREAPP
-                if (jsonWriterFactory is ODataUtf8JsonWriterFactory)
+                if (jsonWriterFactory is ODataJsonWriterFactory)
+                {
+                    // When using the legacy ODataJsonWriterFactory, we wrap the stream in a buffered stream

```

Comment:
Why are we calling it legacy? Can a user use it if they wanted to use it or the plan is to get rid of it all together?

Reply:
If the intention is not to get rid of the default `JsonWriter`.. Are there tests to show how a user can set to use it if they wanted to use it instead of the `ODataUtf8JsonWriter` or we'll never be in that situation? If we'll never be in that situation can we get rid of the `JsonWriter` all together?

Reply:
The user can still use it by injecting the `ODataJsonWriterFactory` (formerly known as `DefaultJsonWriterFactory`) to the service container.

Reply:
Here's an example of a test that shows how the user could configure message writer to use `ODataJsonWriterFactory`: https://github.com/OData/odata.net/pull/2980/files#diff-43dd0673ed97d32066331d297299cab927a8992b7f5cc934f82fecf2d1eae65cR118

Reply:
I used the term `legacy` because it's the original writer, but is no longer default, it's there mostly for backwards compatibility.